### PR TITLE
Dashboard: all pie charts to take care of largeArcFlag

### LINKED ISSUE #1697: Something's wonky with the red pie in dashboard's linkage error pie chart
image

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #1697 (This issue contains the problematic screenshot for the left-most pie chart).

- The strange pie chart in the Linkage Checker error count was caused by largeArcFlag.
- Created pieChartSvg macro that takes care of the flag and applied it to the 4 pie charts.

The screeshot of the fixed version:
Screen Shot 2020-10-14 at 15 39 23

### CODE REVIEW

File: dashboard/src/main/resources/templates/index.ftl [hunk 1]
```diff
@@ -69,71 +69,41 @@
...
            ${plural(localUpperBoundsErrorCount, "does not", "do not")} pick the
            latest versions of all artifacts in their own dependency tree.
-        </td>
-        <td>${globalUpperBoundsErrorCount} out of ${totalArtifacts} artifacts 
-           ${plural(globalUpperBoundsErrorCount, "does not", "do not")} select the
-           most recent version of all artifacts in the BOM.</td>
-        <td>${convergenceErrorCount} out of ${totalArtifacts} artifacts 
-           ${plural(convergenceErrorCount, "fails", "fail")} to converge.
-        </td>
-      </tr>
```
Comment (suztomo) [inline dashboard/src/main/resources/templates/index.ftl:79]: The indentation was incorrect.

File: dashboard/src/main/resources/templates/index.ftl [hunk 2]
```diff
@@ -69,71 +69,41 @@
...
-        <#if ratio gt 0.5>
-          <#assign largeArcFlag = "1">
-        </#if>
-        <#assign endPointX = pieChart.calculateEndPointX(100, 100, 100, ratio)>
-        <#assign endPointY = pieChart.calculateEndPointY(100, 100, 100, ratio)>
-          <svg xmlns="http://www.w3.org/2000/svg" width="${pieSize}" height="${pieSize}">
-            <desc>${globalUpperBoundsErrorCount} out of ${totalArtifacts} artifacts
-                 have global upper bounds errors.</desc>
-            <circle cx="100" cy="100" r="100" stroke-width="3" fill="lightgreen" />
-            <path d="M100,100 v -100 A100,100 0 ${largeArcFlag} 1 ${endPointX}, ${endPointY} z" fill="red" />
```
Comment (suztomo) [inline dashboard/src/main/resources/templates/index.ftl:118]: The right-half of 2 pie charts have largeArcFlag already.

File: dashboard/src/main/resources/templates/index.ftl [hunk 3]
```diff
@@ -69,71 +69,41 @@
...
-          </svg>
-       </td>
-        
-       <td class='pie'>    
-          <#assign ratio = convergenceErrorCount / totalArtifacts >
-          <#assign largeArcFlag = "0">
```
Comment (suztomo) [inline dashboard/src/main/resources/templates/index.ftl:124]: The right-half of 2 pie charts have largeArcFlag already.

File: dashboard/src/main/resources/templates/index.ftl [hunk 4]
```diff
@@ -69,71 +69,41 @@
...
+        </tr>
+        <tr>
+          <td class='pie'>
+            <@pieChartSvg
+                description="${linkageErrorCount} out of ${totalArtifacts} artifacts have linkage
+                    errors."
+                ratio=linkageErrorCount / totalArtifacts />
+          </td>
+          <td class='pie'>
+            <#assign doesNot=plural(localUpperBoundsErrorCount, "does not", "do not")>
```
Comment (suztomo) [inline dashboard/src/main/resources/templates/index.ftl:88]: This function call didn't work well if it's embed in a string. So I extracted it as a variable.

**Review by elharo — APPROVED** [review]

**Pull request merged**